src/main/scala/org/broadinstitute/dsde/firecloud/MethodsClient.scala

```diff
@@ -44,23 +45,41 @@ class MethodsClient(requestContext: RequestContext) extends Actor {
 
     val responseFuture: Future[HttpResponse] = pipeline { Get(FireCloudConfig.Methods.methodsListUrl) }
 
+    def isWhitelisted(origin: Origin): Boolean = {
+      // TODO(dmohs): Check this against a whitelist.
+      true
+    }
+
+    def addCorsHeaders(requestContext: RequestContext): RequestContext = {
```

Comment:
no need to change for this PR. As a comment for the longer term (see DSDEEPB-667), I think we'll want to create a custom spray directive for this, likely built on top of spray's optionalHeaderValueByName directive. The custom directive would look for an Origin header, compare it to a whitelist, and add response headers as appropriate.

See https://github.com/broadinstitute/vault-api/commit/32b3f7f3f11a00ae80c0686152485cf14f2e2690 for an example of wrapping all (most) routes in a custom directive - in that case it was logOpenAMRequest.